code/src/components/Form.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+
+const Form = ({ fetchThoughtList, API_URL }) => {
+  const [newThought, setNewThought] = useState('');
+  const [counter, setCounter] = useState(0);
+
+  const remainingCharacters = 140 - counter;
+
+  const handleNewThoughtChange = (e) => {
+    setNewThought(e.target.value);
+    setCounter(e.target.value.length);
+  };
+
+  //   form submitted
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ message: newThought })
+    };
+    fetch(API_URL, options)
+      .then((res) => res.json())
+      .then(() => fetchThoughtList())
+      .catch((err) => console.error(err));
+    setNewThought('');
+    setCounter(0);
+  };
+
+  return (
+    <form onSubmit={onFormSubmit}>
+      <label htmlFor='newThought'>
+        What makes you happy right now?{' '}
+        <textarea
+          className={
+            counter < 6 || counter > 140 ? 'disabled-textarea' : 'textarea'
```

Comment:
I like this validation , looks natural and logical . You have a countdown , so when it is 0 out of 140 )) you are done lol.